File: L16AdobeEasy/BaseballGame.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution
{
public:
    int calPoints(vector<string> &ops)
    {
        vector<int> ans;

        for (int i = 0; i < ops.size(); i++)
        {
            if (ops[i] == "C")
            {
                if (!ans.empty())
                    ans.pop_back(); // Remove last valid score
            }
            else if (ops[i] == "D")
            {
                if (!ans.empty())
                    ans.push_back(2 * ans.back()); // Double last score
            }
            else if (ops[i] == "+")
            {
                if (ans.size() >= 2)
                {
                    int a = ans[ans.size() - 1];
                    int b = ans[ans.size() - 2];
                    ans.push_back(a + b); // Sum of last two scores
                }
            }
            else
            {
                ans.push_back(stoi(ops[i])); // Convert string to int and add to list
            }
        }

        int sum = 0;
        for (int i = 0; i < ans.size(); i++)
        {
            sum += ans[i]; // Sum all valid scores
        }

        return sum;
    }
};

int main()
{
    Solution sol;
    vector<string> ops = {"5", "2", "C", "D", "+"}; // Example test case
    cout << "Final Score: " << sol.calPoints(ops) << endl;
    return 0;
}

File: L16AdobeEasy/BaseballGame.java

import java.util.ArrayList;
import java.util.List;

public class BaseballGame {

    public int calPoints(List<String> ops) {
        List<Integer> ans = new ArrayList<>();

        for (String op : ops) {
            if (op.equals("C")) {
                if (!ans.isEmpty()) {
                    ans.remove(ans.size() - 1);  // Remove last valid score
                }
            } else if (op.equals("D")) {
                if (!ans.isEmpty()) {
                    ans.add(2 * ans.get(ans.size() - 1));  // Double last score
                }
            } else if (op.equals("+")) {
                if (ans.size() >= 2) {
                    int a = ans.get(ans.size() - 1);
                    int b = ans.get(ans.size() - 2);
                    ans.add(a + b);  // Sum of last two scores
                }
            } else {
                ans.add(Integer.valueOf(op));  // Convert string to integer
            }
        }

        int sum = 0;
        for (int score : ans) {
            sum += score;  // Calculate total score
        }

        return sum;
    }

    public static void main(String[] args) {
        BaseballGame game = new BaseballGame();
        List<String> ops = new ArrayList<>();
        ops.add("5");
        ops.add("2");
        ops.add("C");
        ops.add("D");
        ops.add("+");

        System.out.println("Final Score: " + game.calPoints(ops));
    }
}
